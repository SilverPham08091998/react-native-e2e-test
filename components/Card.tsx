import { ViewStyle, ViewProps, View } from 'react-native';
import { ThemeProps, useThemeColor } from './Themed';

const roundeds = {
  none: { borderRadius: 0 } as ViewStyle,
  light: { borderRadius: 5 } as ViewStyle,
  normal: { borderRadius: 10 } as ViewStyle,
  medium: { borderRadius: 17 } as ViewStyle,
  huge: { borderRadius: 56 } as ViewStyle,
};
type Roundeds = keyof typeof roundeds;
export type CardProps = ThemeProps &
  ViewProps & {
    rounded?: Roundeds;
  };
export function Card(props: CardProps) {
  const { rounded = 'none', style, lightColor, darkColor, ...otherViewProps } = props;
  const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, 'card');
  const cardStyle = [roundeds[rounded], { backgroundColor }, style];
  return <View style={cardStyle} {...otherViewProps} />;
}
