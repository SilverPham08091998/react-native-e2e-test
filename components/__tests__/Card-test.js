import * as React from 'react';
import renderer from 'react-test-renderer';

import { Card } from '../Card';

it(`renders correctly`, () => {
  const tree = renderer.create(<Card rounded="medium" lightColor='red' darkColor='blue'>Snapshot test!</Card>).toJSON();

  expect(tree).toMatchSnapshot();
});
