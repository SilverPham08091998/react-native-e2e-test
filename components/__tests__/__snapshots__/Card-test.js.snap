// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "borderRadius": 17,
      },
      {
        "backgroundColor": "red",
      },
      undefined,
    ]
  }
>
  Snapshot test!
</View>
`;
